src/index.ts

```diff
@@ -74,12 +73,18 @@ export type Component<So, Si> = (sources: So, ...rest: Array<any>) => Si;
  * original `dataflowComponent` function, takes `sources` and returns `sinks`.
  * @function isolate
  */
-function isolate<So, Si>(component: Component<So, Si>, scope: string = newScope()): Component<So, Si> {
+function isolate<So, Si>(component: Component<So, Si>, scope: any = newScope()): Component<So, Si> {
   checkIsolateArgs(component, scope);
+  let convertedScope: string;
+  if (typeof scope === 'string') {
+    convertedScope = scope;
+  }else if (typeof scope !== 'string') {
```

Comment:
This could be just an `else`, not `else if`. In fact, it could be just a ternary operator:
```js
let convertedScope: string = typeof scope === 'string' ? scope : scope.toString();
```